### PR TITLE
File viewer v2

### PR DESCRIPTION
**Pull request by btzr-io**
### Changes (WIP)
- [x] Closes: Better preview for content #620
- [x] Closes: Use router pattern for rendering file viewer ([#1544](https://github.com/lbryio/lbry-desktop/pull/1544))
- [x]  Temp solution for:  Add csv and json viewer ([#1410](https://github.com/lbryio/lbry-desktop/pull/1410))
- [x] Markdown viewer. (md, markdown)
- [x]  Plain-text files and code snippets viewer (txt, json, csv, xml, log, yml).
- [x] Docx viewer (text only)
- [x] Html viewer. (iframe sandbox)
- [x] Implement syntax highlighter.
- [x] Fix description field fullscreen styles https://github.com/lbryio/lbry-desktop/pull/1826/commits/77cfcc84a81f1e629895d7f2019a318d1a11de30.
- [x] Fix `CodeMirror` context menu https://github.com/lbryio/lbry-desktop/pull/1671/commits/be9c16cbdec789b557da118b1fc09b950be7171a.
- [x] Choose a `CodeMirror` theme (light / dark).
- [x] Refactoring of fileSource logic.
- [x] Problem with search auto-complete menu when scrolling over file viewer #1847 

> lbry://pdffile#1ec33dbcdf61c051b136354a9640f5dc25b04b96

> lbry://csvfile#3eb2ae7de726f21318fc61c6794960f7fc4d81da 

> lbry://json#5fd9811240848eeae120166d8c57fc8e81dc6497

> lbry://txtfile#13475689e84cd6c3ef43a279c0e8bfa6ed5c312f

> lbry://index#6e477eded9468613c9c5b45eb58662ea302bccea

> lbry://mdfile#24e737e03df62930ac4bd44151c818b3933a0381

> bry://javascriptfile#c286d02c447aecb9cbd83b6f3ee6811e8fcc3d68

### CODE REVIEW

**Comment by skhameneh** [comment]
Looks good, nice work!
I'll need to take a closer look and test this out; @seanyesmunt also had some MD changes as well that I need to test and check for conflicts.

**Comment by btzr-io** [comment]
https://github.com/codemirror/CodeMirror/issues/5484

**Comment by kauffj** [comment]
FYI @btzr-io @seanyesmunt is on vacation until Wednesday

**Comment by seanyesmunt** [comment]
Just added this to test lbry://javascriptfile#c286d02c447aecb9cbd83b6f3ee6811e8fcc3d68

File: src/renderer/scss/component/_file-render.scss [hunk 1]
```diff
@@ -9,17 +9,80 @@
...
+
+  .CodeMirror-line {
+    padding-left: 16px;
+  }
+
+  .CodeMirror-gutters {
+    border-right: 1px solid var(--color-divider);
+    background: var(--color-bg-alt);
+    padding-right: 8px;
+    z-index: 3;
```
Comment (seanyesmunt) [inline src/renderer/scss/component/_file-render.scss]: Whoops forgot to add this comment. This `z-index` seems a little weird. Is it needed?

Comment (btzr-io) [reply src/renderer/scss/component/_file-render.scss]: not sure let me check, probably not :+1:

**Comment by btzr-io** [comment]
@seanyesmunt ok done :+1:

**Comment by tzarebczan** [comment]
On lbry://@jiggytom#e48d2b50501159034f68d53321f67b8aa5b1d771/epubfile - a non-supported file, I'm missing a view button. Not sure if it should still say view, but lbry://zipfile does and it's an unsupported file. This isn't related to this PR, but thought i'd mention it as we are touching that code. 

Otherwise, I tested a bunch of the videos and things look good! Thanks again for the contribution @btzr-io !

**Pull request merged**